tests/ui/transmutability/region-infer.stderr

```diff
@@ -2,7 +2,7 @@ error[E0277]: `()` cannot be safely transmuted into `W<'_>` in the defining scop
   --> $DIR/region-infer.rs:20:5
    |
 LL |     test();
-   |     ^^^^ `W<'_>` does not have a well-specified layout
```

Comment:
I *think* this change is expected because the previous error was because we weren't supporting references.

This is what is in the test
```rs
#[repr(C)]
struct W<'a>(&'a ());
```

Reply:
@jswrenn do you agree with that^?

Reply:
Thinking about it more, the new error is
```
The size of `()` is smaller than the size of `W<'_>`
```
which makes sense because `W` needs to store the ptr. So `()` does not hold enough data to be able to transmute to `W`. Just going to resolve this comment for now!